components/dashboard/src/data/prebuilds/prebuild-logs-emitter.ts

```diff
@@ -145,29 +145,40 @@ function streamPrebuildLogs(
         };
 
         let response: Response | undefined = undefined;
+        let abortController: AbortController | undefined = undefined;
         let reader: ReadableStreamDefaultReader<Uint8Array> | undefined = undefined;
         try {
+            abortController = new AbortController();
+            disposables.push({
+                dispose: async () => {
+                    abortController?.abort();
+                    await reader?.cancel();
+                },
+            });
             console.debug("fetching from streamUrl: " + streamUrl);
             response = await fetch(streamUrl, {
                 method: "GET",
-                cache: "no-cache",
+                cache: "reload",
                 credentials: "include",
-                keepalive: true,
                 headers: {
                     TE: "trailers", // necessary to receive stream status code
                 },
+                signal: abortController.signal,
                 redirect: "follow",
             });
             reader = response.body?.getReader();
             if (!reader) {
                 await retryBackoff("no reader");
                 return;
             }
-            disposables.push({ dispose: () => reader?.cancel() });
 
             const decoder = new TextDecoder("utf-8");
             let chunk = await reader.read();
             while (!chunk.done) {
+                if (disposables.disposed) {
+                    // stop reading when disposed
```

Comment:
this means we don't have to read the whole body anymore if we dispose in the midst of it.